src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala

```diff
@@ -0,0 +1,410 @@
+
+    resourceMatchOpt match {
+      case Some(resourceMatch) =>
+        build(podDefinition, offer, newTaskId, config, resourceMatch)
+      case _ =>
+        None
+    }
+  }
+
+  private[this] def build(
+    podDefinition: PodDefinition,
+    offer: mesos.Offer,
+    newTaskId: PathId => Instance.Id,
+    config: MarathonConf,
+    resourceMatch: ResourceMatcher.ResourceMatch
+  ): Some[(mesos.ExecutorInfo, mesos.TaskGroupInfo, Seq[Option[Int]])] = {
+    // TODO: probably set unique ID for each task
+    val taskId = newTaskId(podDefinition.id)
+
+    val executorInfo = computeExecutorInfo(podDefinition, taskId)
+
+    val taskGroup = mesos.TaskGroupInfo.newBuilder
+
+    podDefinition.containers
+      .map(computeTaskInfo(_, podDefinition, offer, taskId, config, resourceMatch))
+      .foreach(taskGroup.addTasks)
+
+    Some((executorInfo.build, taskGroup.build, resourceMatch.hostPorts))
+  }
+
+  private[this] def computeTaskInfo(
+    container: raml.MesosContainer,
+    podDefinition: PodDefinition,
+    offer: mesos.Offer,
+    taskId: Instance.Id,
+    config: MarathonConf,
+    resourceMatch: ResourceMatcher.ResourceMatch
+  ): mesos.TaskInfo.Builder = {
+    val builder = mesos.TaskInfo.newBuilder
+      .setName(container.name)
+      .setTaskId(taskId.mesosTaskId)
+      .setSlaveId(offer.getSlaveId)
+
+    builder.addResources(scalarResource("cpus", container.resources.cpus))
+    builder.addResources(scalarResource("mem", container.resources.mem))
+    builder.addResources(scalarResource("disk", container.resources.disk))
+    builder.addResources(scalarResource("gpus", container.resources.gpus.toDouble))
+
+    val labels = podDefinition.labels ++ container.labels.map(_.values).getOrElse(Map.empty[String, String])
+
+    if (labels.nonEmpty)
+      builder.setLabels(mesos.Labels.newBuilder.addAllLabels(labels.map {
+        case (key, value) =>
+          mesos.Label.newBuilder.setKey(key).setValue(value).build
+      }.asJava))
+
+    val envPrefix: Option[String] = config.envVarsPrefix.get
+
+    val commandInfo = computeCommandInfo(
+      podDefinition,
+      taskId,
+      container,
+      offer.getHostname,
+      resourceMatch.hostPorts,
+      envPrefix)
+
+    builder.setCommand(commandInfo)
+
+    val containerInfo = computeContainerInfo(podDefinition.podVolumes, container)
+
+    builder.setContainer(containerInfo)
+
+    container.healthCheck.foreach { healthCheck =>
+      builder.setHealthCheck(computeHealthCheck(healthCheck, container.endpoints))
+    }
+
+    builder
+  }
+
+  private[this] def computeExecutorInfo(
+    podDefinition: PodDefinition,
+    taskId: Instance.Id): mesos.ExecutorInfo.Builder = {
+    val executorID = mesos.ExecutorID.newBuilder.setValue(f"marathon-${taskId.idString}")
+
+    val executorInfo = mesos.ExecutorInfo.newBuilder
+      .setType(mesos.ExecutorInfo.Type.DEFAULT)
+      .setExecutorId(executorID)
```

Comment:
looks like the codes *does* use instanceId. The variable is poorly named.
s/taskId/instanceId/

On Mon, Sep 12, 2016 at 8:17 AM, Matthias Veit <notifications@github.com>
wrote:

> In src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala
> <https://github.com/mesosphere/marathon/pull/4353#discussion_r78361079>:
>
> > +
> > +    container.healthCheck.foreach { healthCheck =>
> > +      builder.setHealthCheck(computeHealthCheck(healthCheck, container.endpoints))
> > +    }
> > +
> > +    builder
> > +  }
> > +
> > +  private[this] def computeExecutorInfo(
> > +    podDefinition: PodDefinition,
> > +    taskId: Instance.Id): mesos.ExecutorInfo.Builder = {
> > +    val executorID = mesos.ExecutorID.newBuilder.setValue(f"marathon-${taskId.idString}")
> > +
> > +    val executorInfo = mesos.ExecutorInfo.newBuilder
> > +      .setType(mesos.ExecutorInfo.Type.DEFAULT)
> > +      .setExecutorId(executorID)
>
> I would suggest to pick the instanceId as value here.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4353/files/0a69499a62feb921a20eff42c484a997f004cd01#r78361079>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACPVLP2b9ZmOEgHjP3ClSR5lqV6C9aoBks5qpULpgaJpZM4J6bJ5>
> .
>